### PR TITLE
feat: modules cloudevents, asyncapi and yaml added

### PR DESCRIPTION
**Pull request by sliesche**
Not marked as **breaking change** since features were only added not changed

Notable changes to dropwizard:

- sda-commons-cloudevents: tests adapted to Spring Boot including usage of spring kafka JsonDeserializer
- sda-commons-asyncapi: uses now Spring Boot injection instead of static methods, split into separate classes, prototype scope for JsonBuilder, AsyncBuilder. Spring OOTB ObjectMapper is used

### CODE REVIEW

File: sda-commons-asyncapi/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# SDA Commons Shared AsyncAPI
```
Comment (christopher-cudennec) [inline sda-commons-asyncapi/README.md:1]: Oh, more than I expected. I thought about using [SpringWolf](https://github.com/springwolf/springwolf-core) for AsyncAPI documentation. I got this as another missing feature for Spring Boot Commons in the back of my head. Let's discuss this in the team.

File: sda-commons-cloudevents/src/test/java/org/sdase/commons/server/cloudevents/ContractCloudEventTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    classes = KafkaTestApp.class,
+    properties = {
+      "management.server.port=0",
+      "spring.kafka.bootstrap-servers=${spring.embedded.kafka.brokers}",
+      "spring.kafka.producer.value-serializer=org.springframework.kafka.support.serializer.JsonSerializer"
+    })
+@EmbeddedKafka(
+    partitions = 1,
+    brokerProperties = {"listeners=PLAINTEXT://localhost:9092", "port=9092"})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
```
Comment (christopher-cudennec) [inline sda-commons-cloudevents/src/test/java/org/sdase/commons/server/cloudevents/ContractCloudEventTest.java:48]: Do you know the reason why we use `DirtiesContext` here? Does every Kafka test need/use that annotation?

Comment (sliesche) [reply sda-commons-cloudevents/src/test/java/org/sdase/commons/server/cloudevents/ContractCloudEventTest.java:48]: My understanding is that the EmbeddedKafka annotation adds an embedded Kafka to the test context which is cleaned up using DirtiesContext. used in the example provided by the spring boot kafka [documentation](https://docs.spring.io/spring-kafka/reference/html/#embedded-kafka-annotation) If not used there is a chance that some spring caching magic uses the same kafka broker in other tests.

Actually i think that both Kafka tests in cloud events could use the same broker which would speed up test execution. The documentation provides an [example](https://docs.spring.io/spring-kafka/reference/html/#same-broker-multiple-tests), although it would be easier to just put the two test in the same test class.

However, these test provide an easy to understand and safe example that will be most likely used in other tests

File: sda-commons-cloudevents/build.gradle [hunk 3]
```diff
@@ -0,0 +1,12 @@
+dependencies {
+
+  api ('org.springframework.boot:spring-boot-starter-validation')
```
Comment (JoergSiebahn) [inline sda-commons-cloudevents/build.gradle:3]: Do we really need this dependency for production? I would prefer to only include `javax.validation` which is referenced in imports.

File: sda-commons-asyncapi/README.md (hunk 1 above)
Comment (JoergSiebahn) [reply sda-commons-asyncapi/README.md:1]: I would agree that the AsyncAPI generation is beyond the scope of the current task. It also bloats this PR quite a lot. However, it is another dependency that is currently taken from our Dropwizard library and mixes up the dependency tree in the Spring services.

I heard mixed opinions about Spring Wolf from customers in our microservice context. In general the result seem to be _good enough_ but not completely satisfying. The AsyncAPIs we can produce with our current implementation are describing the API very well. There are still some edge cases that can't be documented as good as it should be (e.g. describing object type fields separately from the object definition). Also the library is quite old and should only be used in test code.

I think, it's not too bad to have a solution in place right away. But I would prefer further research to find a better (and actively maintained) solution. That may be Spring Wolf if we find a way to create a satisfying result that is easy to use. What do you think about marking the AsyncAPI module as a workaround alpha solution that is subject to be replaced without further notice?

File: sda-commons-yaml/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# SDA Commons Shared YAML
```
Comment (JoergSiebahn) [inline sda-commons-yaml/README.md:1]: Do we really need this module? Isn't there something available from Spring? I would like to avoid maintaining more code than necessary.

File: sda-commons-asyncapi/README.md (hunk 1 above)
Comment (sliesche) [reply sda-commons-asyncapi/README.md]: Yaml module removed.

Updated Async Readme

> ## ⚠️ Experimental ⚠️
>
> ### Please be aware that SDA SE is likely to change or remove this artifact in the future

File: sda-commons-yaml/README.md (hunk 4 above)
Comment (sliesche) [reply sda-commons-yaml/README.md]: module removed

`com.fasterxml.jackson.dataformat.yaml.YAMLMapper` used in asyncapi module

File: sda-commons-asyncapi/src/test/java/org/sdase/commons/spring/boot/asyncapi/internal/JsonSchemaEmbedderTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  //
+  //    assertThat(result.at("/definitions").fieldNames())
+  //        .toIterable()
+  //        .containsExactlyInAnyOrder("Person", "simpleobjectsimplereferencedyaml");
+  //    assertThat(result.at("/definitions/simpleobjectsimplereferencedyaml").isMissingNode())
+  //        .isFalse();
+  //    assertThat(result.at("/definitions/address").isMissingNode()).isTrue();
+  //    assertThat(result.at("/definitions/Person/properties/address/$ref").asText())
+  //        .isEqualTo("https://sda.se");
+  //  }
```
Comment (JoergSiebahn) [inline sda-commons-asyncapi/src/test/java/org/sdase/commons/spring/boot/asyncapi/internal/JsonSchemaEmbedderTest.java:74]: NIT: Could we remove this code, enable the test or disable the test?

File: sda-commons-cloudevents/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+> The bundle is still in an early state. APIs are open for change.
+>
+
+## Introduction
+
+CloudEvents is a general standard that can be used in combination with your favourite eventing tool
+like ActiveMQ or Kafka. The CloudEvents specification defines concrete bindings to define how the
+general specification should be applied to a specific tool. This module brings you
+the [Kafka protocol binding](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/kafka-protocol-binding.md)
+.
```
Comment (JoergSiebahn) [inline sda-commons-cloudevents/README.md:16]: NIT

```suggestion
the [Kafka protocol binding](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/bindings/kafka-protocol-binding.md).
```

**Review by JoergSiebahn — APPROVED** [review]

File: sda-commons-cloudevents/README.md (hunk 6 above)
Comment (sliesche) [reply sda-commons-cloudevents/README.md]: updated

File: sda-commons-asyncapi/src/test/java/org/sdase/commons/spring/boot/asyncapi/internal/JsonSchemaEmbedderTest.java (hunk 5 above)
Comment (sliesche) [reply sda-commons-asyncapi/src/test/java/org/sdase/commons/spring/boot/asyncapi/internal/JsonSchemaEmbedderTest.java]: good find! i deactivated it to fix something

enabled test

**Review by JoergSiebahn — APPROVED** [review]
